src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -468,17 +484,42 @@ private BoundExpression MakeConversionNode(BoundExpression rewrittenOperand, Typ
                      rewrittenOperand.Type.SpecialType != SpecialType.System_DateTime)
                 {
                     // error CS0029: Cannot implicitly convert type '{0}' to '{1}'
-                    _diagnostics.Add(
+                    diagnostics.Add(
                         ErrorCode.ERR_NoImplicitConv,
                         rewrittenOperand.Syntax.Location,
                         rewrittenOperand.Type,
                         rewrittenType);
                 }
+            }
 
-                return _factory.NullOrDefault(rewrittenType);
+            return conversion;
+        } 
+
+        private static BoundExpression MakeConversionForIOperation(
+            BoundExpression operand, 
+            TypeSymbol type, 
+            SyntaxNode syntax, 
+            CSharpCompilation compilation, 
+            DiagnosticBag diagnostics, 
+            bool @checked, 
+            bool acceptFailingConversion = false)
+        {
+            Conversion conversion = MakeConversion(operand, type, compilation, diagnostics, acceptFailingConversion);
+
+            if (conversion.Kind == ConversionKind.Identity)
+            {
+                return operand;
             }
 
-            return MakeConversionNode(rewrittenOperand.Syntax, rewrittenOperand, conversion, rewrittenType, @checked);
+            return new BoundConversion(
+                            syntax,
+                            operand,
+                            conversion,             
+                            @checked: @checked,
+                            explicitCastInCode: false,
+                            constantValueOpt: default(ConstantValue),
```

Comment:
>constantValueOpt: default(ConstantValue) [](start = 28, length = 40)

Are we using this function to convert default parameter values? We might want to do constant folding for them.